RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc

```diff
@@ -63,6 +73,27 @@ void MkFitEventOfHitsProducer::produce(edm::StreamID iID, edm::Event& iEvent, co
   auto eventOfHits = std::make_unique<mkfit::EventOfHits>(mkFitGeom.trackerInfo());
   mkfit::StdSeq::Cmssw_LoadHits_Begin(*eventOfHits, {&pixelHits.hits(), &stripHits.hits()});
 
+  std::vector<mkfit::DeadVec> deadvectors(mkFitGeom.layerNumberConverter().nLayers());
+  const auto& siStripQuality = iSetup.getData(qualityToken_);
+  const auto& trackerGeom = iSetup.getData(geomToken_);
+  const auto& badStrips = siStripQuality.getBadComponentList();
+  for (const auto& bs : badStrips) {
+    const auto& surf = trackerGeom.idToDet(DetId(bs.detid))->surface();
+    const DetId detid(bs.detid);
+    const auto subdet = detid.subdetId();
+    const auto layer = mkFitGeom.topology()->layer(detid);
+    const auto isStereo = mkFitGeom.topology()->isStereo(detid);
+    bool isBarrel = (mkFitGeom.topology()->side(detid) == static_cast<unsigned>(TrackerDetSide::Barrel));
+    bool isPlusSide = (mkFitGeom.topology()->side(detid) == static_cast<unsigned>(TrackerDetSide::PosEndcap));
+    const auto ilay = mkFitGeom.layerNumberConverter().convertLayerNumber(subdet, layer, false, isStereo, isPlusSide);
```

Comment:
This (and many of the intermediate variables) could be replaced with
```suggestion
    const auto ilay = mkFitGeom.mkFitLayerNumber(detid);
```

Reply:
thanks, I am updating the PR